onnxruntime/core/mlas/lib/cast.cpp

```diff
@@ -71,16 +71,16 @@ MlasConvertHalfToFloatBufferInParallel(
             ThreadPool, Count,
             // Tensor Op Cost
             {
-                static_cast<double>(Count * sizeof(MLAS_FP16)),  // Size of no. of elements in bytes to be loaded
-                static_cast<double>(Count * sizeof(float)),      // Size of no. of elements in bytes to be stored
-                static_cast<double>(num_compute_cycles),         // No. of compute cycles required for the tensor op
+                gsl::narrow<double>(Count * sizeof(MLAS_FP16)),  // Size of no. of elements in bytes to be loaded
```

Comment:
generally, we should use `onnxruntime::narrow` from include/onnxruntime/core/common/narrow.h. it also supports builds where exceptions are disabled.